lambda-function-sqs-report-batch-item-failures/example.mjs

```diff
@@ -0,0 +1,20 @@
+export const handler = async (event, context) => {
+    const batchItemFailures = [];
+
+    for (const record of event.Records) {
+        try {
+            await processMessageAsync(record, context);
+        } catch (error) {
+            batchItemFailures.push({ itemIdentifier: record.messageId });
+        }
+    }
+
+    return { batchItemFailures };
+};
+
+async function processMessageAsync(record, context) {
+    if (record.body && record.body.includes("error")) {
+        throw new Error("There is an error in the SQS Message.");
+    }
+    console.log(`Processed message: ${record.body}`);
```

Comment:
Use context.log here?

Reply:
I've never seen 'context' used like this before, and it didn't log anything to CloudWatch, while 'console.log' did. As far as documentation goes, 'context' is a read-only object that provides the information about the invocation, function, and execution environment.